src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+        }
+
+        const base64String = options?.base64?.replace('\r', '')?.replace('\n', '');
+        let successful: boolean;
+        if (base64String != undefined) {
+          // read in peer info string
+          const { publicKey, connectedAddress, relayPublicKey } = PeerInfo.parseB64(base64String);
+          successful = await client.updatePeer(
+            nodePath,
+            currentNode ? undefined : publicKey,
+            currentNode,
+            connectedAddress?.host,
+            connectedAddress?.port,
+            relayPublicKey,
+          );
+        } else {
+          successful = await client.updatePeer(
+            nodePath,
+            publicKey,
+            currentNode,
+            options.connectedHost,
+            options.connectedPort,
+            relayPublicKey,
+          );
+        }
+
+        if (successful) {
+          pkLogger('peer info was successfully updated', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('something went wrong, peer info could not be updated', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+function makeGetPeerInfoCommand() {
+  return new commander.Command('get')
+    .description('get the peer info for a particular public key')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-b, --base64', 'output peer info as a base64 string')
+    .option('-c, --current-node', 'only list the peer information for the current node, useful for sharing')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const currentNode: boolean = options.currentNode;
+        const publicKeyPath = options.publicKey;
+        const base64 = options.base64;
+        // read in publicKey if it exists
+        let publicKey: string | undefined;
+        if (publicKeyPath) {
+          publicKey = fs.readFileSync(publicKeyPath).toString();
+        }
+
+        const peerInfo: PeerInfo = await client.getPeerInfo(nodePath, currentNode, publicKey);
+
+        if (base64) {
+          pkLogger(peerInfo.toStringB64(), PKMessageType.SUCCESS);
+        } else {
+          pkLogger('Peer Public Key:', PKMessageType.INFO);
+          pkLogger(peerInfo.publicKey, PKMessageType.SUCCESS);
+
+          pkLogger('Peer Connected Address:', PKMessageType.INFO);
+          pkLogger(peerInfo.connectedAddress?.toString() ?? '', PKMessageType.SUCCESS);
+
+          pkLogger('Relay Public Key:', PKMessageType.INFO);
+          pkLogger(peerInfo.relayPublicKey ?? '', PKMessageType.SUCCESS);
```

Comment:
Yea so here is exactly what I'm talking about. You need to separate logging from outputing to STDOUT. The logger should ALWAYS go to STDERR.

For outputting to STDOUT. I use a custom `format` function. And then just write directly to STDOUT.

I may even refer to the fd directly and not use `console.log` at all. This is because sometimes you need to output without the trailing newline character. And sometimes you are outputting bytes, or structured objects that need to be formatted!